File: eRents.UI/e_rents_mobile/lib/core/models/payment.dart
class Payment {
  final int paymentId;
  final int? tenantId;
  final int? propertyId;
  final double amount;
  final DateTime? datePaid;
  final String? paymentMethod;
  final String? paymentStatus;
  final String? paymentReference;

  Payment({
    required this.paymentId,
    this.tenantId,
    this.propertyId,
    required this.amount,
    this.datePaid,
    this.paymentMethod,
    this.paymentStatus,
    this.paymentReference,
  });

  factory Payment.fromJson(Map<String, dynamic> json) {
    return Payment(
      paymentId: json['paymentId'],
      tenantId: json['tenantId'],
      propertyId: json['propertyId'],
      amount: json['amount'].toDouble(),
      datePaid: json['datePaid'] != null ? DateTime.parse(json['datePaid']) : null,
      paymentMethod: json['paymentMethod'],
      paymentStatus: json['paymentStatus'],
      paymentReference: json['paymentReference'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'paymentId': paymentId,
      'tenantId': tenantId,
      'propertyId': propertyId,
      'amount': amount,
      'datePaid': datePaid?.toIso8601String(),
      'paymentMethod': paymentMethod,
      'paymentStatus': paymentStatus,
      'paymentReference': paymentReference,
    };
  }
}

File: eRents.Domain/Models/Payment.cs
using System;
using System.Collections.Generic;
using eRents.Domain.Shared;

namespace eRents.Domain.Models;

public partial class Payment : BaseEntity
{
    public int PaymentId { get; set; }

    public int? TenantId { get; set; }

    public int? PropertyId { get; set; }

    public int? BookingId { get; set; }

    public int? SubscriptionId { get; set; }

    public decimal Amount { get; set; }

    public string? Currency { get; set; }

    public string? PaymentMethod { get; set; }

    public string? PaymentStatus { get; set; }

    public string? PaymentReference { get; set; }

    // Stripe payment fields
    public string? StripePaymentIntentId { get; set; }
    
    public string? StripeChargeId { get; set; }

    // Additional fields for refund support
    public int? OriginalPaymentId { get; set; }
    
    public string? RefundReason { get; set; }
    
    public string? PaymentType { get; set; } = "BookingPayment"; // BookingPayment, Refund

    // Navigation Properties
    public virtual Booking? Booking { get; set; }

    public virtual Property? Property { get; set; }

    public virtual Tenant? Tenant { get; set; }
    
    public virtual Subscription? Subscription { get; set; }
    
    public virtual Payment? OriginalPayment { get; set; }
    
    public virtual ICollection<Payment> Refunds { get; set; } = new List<Payment>();
}
